hypershift-operator/controllers/hostedcluster/internal/platform/kubevirt/kubevirt.go

```diff
@@ -29,7 +29,7 @@ func (p Kubevirt) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, cre
 	kubevirtCluster := &capikubevirt.KubevirtCluster{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: controlPlaneNamespace,
-			Name:      hcluster.Spec.InfraID,
+			Name:      hcluster.Name,
```

Comment:
This change breaks Kubevirt platform cluster creation
In order to update the ProviderID on the Nodes, capk looks for the kubeconfig secret in the cluster namespace
it is looking for secret with name with the following format: `<kubevirtcluster_name>-kubeconfig`
This enforce us to set the infraID as the kubevirtcluster name

Also, the cluster resource is created with name=infraID, so in my opinion it also make sense to create the kubevirtcluster resource with the same name

Reply:
Now I see you also changed the cluster name not to have the infraID in the name
I will test kubevirt platform creation manually on this branch

Reply:
ok thanks, All you mention here seems cometic and implementation details. My question above is conceptually about API semantic and platform scenarios. From what I'm hearing kubevirt does not need an infraID user input.

Reply:
Yes, its not needed